packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -934,11 +938,15 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
     }
 
     shouldShowInferFromFile(): boolean {
-        const { domain, showInferFromFile } = this.props;
-        return showInferFromFile && domain.fields.size === 0;
+        const { domain, domainFormDisplayOptions } = this.props;
+        return !domainFormDisplayOptions.hideInferFromFile && domain.fields.size === 0;
+    }
+
+    shouldShowImportExport(): boolean {
+        return !this.props.domainFormDisplayOptions.hideImportExport;
```

Comment:
Null check for domainFormDisplayOptions?

Reply:
Good idea. Added.